### PR TITLE
Add Python auth example

### PR DESCRIPTION
**Pull request by soltanmm**
cc @LisaFC @jboeuf

### CODE REVIEW

**Comment by nathanielmanistaatgoogle** [comment]
Looks good; reassigning to @jboeuf for his review.

File: docs/guides/auth.md [hunk 1]
```diff
@@ -175,6 +175,23 @@ var channel = new Channel("myservice.example.com", channelCredentials);
...
+import helloworld_pb2
+
+# Base case - No encryption
+channel = implementations.insecure_channel('localhost', 50051)
+stub = helloworld_pb2.beta_create_Greeter_stub(channel)
+...
+
+# With server authentication SSL/TLS
+creds = implementations.ssl_channel_credentials(open('roots.pem').read(), None, None)
+channel = implementations.secure_channel('localhost', 50051, creds)
```
Comment (jboeuf) [inline docs/guides/auth.md]: please replace `'localhost', 50051` by `'greeter.googleapis.com', 443`.

File: docs/guides/auth.md [hunk 2]
```diff
@@ -326,3 +343,31 @@ $opts = [
...
+    credentials = oauth2client.client.GoogleCredentials.get_application_default()
+    scoped_credentials = credentials.create_scoped([scope])
+  except Exception as error:
+    callback([], error)
+    return
+  callback([('authorization', 'Bearer %s' % scoped_credentials.get_access_token().access_token)], None)
+
+auth_creds = implementations.metadata_plugin_credentials(oauth2token_credentials)
+channel_creds = implementations.composite_channel_credentials(transport_creds, auth_creds)
+channel = implementations.secure_channel('localhost', 50051, channel_creds)
```
Comment (jboeuf) [inline docs/guides/auth.md]: please replace `'localhost', 50051` by `'greeter.googleapis.com', 443`.

File: docs/guides/auth.md [hunk 3]
```diff
@@ -326,3 +343,31 @@ $opts = [
...
+###Authenticating with Google (Python)
+
+####Base case - No encryption/authentication
+```python
+channel = implementations.insecure_channel('localhost', 50051)
+stub = helloworld_pb2.beta_create_Greeter_stub(channel)
+...
+```
+
+###Authenticate using OAuth2 token (legacy approach)
```
Comment (jboeuf) [inline docs/guides/auth.md]: so, do we only have the legacy approach in python?

Comment (soltanmm) [inline docs/guides/auth.md]: Scopeless credentials are not implemented in oauth2client. :-/

Comment (jboeuf) [inline docs/guides/auth.md]: :(. @anthmgoogle is there a timeline for this support? 

@soltanmm can you please make the fixes above and I'll LGTM (and reassign @LisaFC). Thanks!

Comment (anthmgoogle) [inline docs/guides/auth.md]: Do you mean JWT credentials? @nathanielmanistaatgoogle, @tbetbetbe, I thought this was already implement in oauth2client.

Comment (jboeuf) [inline docs/guides/auth.md]: @nathanielmanistaatgoogle @soltanmm @tbetbetbe can someone reply to @anthmgoogle ?

Comment (nathanielmanistaatgoogle) [inline docs/guides/auth.md]: @anthmgoogle: I think what happened was that @soltanmm and I were unclear on the work to be done, @tbetbetbe volunteered to take it on, and then it never got pushed back to us when he took on other work. I think @soltanmm would be willing to return to it if the questions he had when last working on it were answered.

Let's hash out the specifics of the work in an issue in https://github.com/google/oauth2client and then implement it?

Comment (nathanielmanistaatgoogle) [inline docs/guides/auth.md]: https://github.com/google/oauth2client/issues/377.

Comment (jboeuf) [inline docs/guides/auth.md]: Sorry. Do you mean that JWT access was never implemented in the python oauth2client? Or is it the default application credentials that was never hooked up with it for gRPC?

Comment (jboeuf) [inline docs/guides/auth.md]: @jayantkolhe FYI...

Comment (anthmgoogle) [inline docs/guides/auth.md]: I think this did fall through the cracks. For some reason I thought it was done, but in retrospect I remember reviewing the Go, Ruby and PHP implementations but not this one. We did have a tracking bug https://github.com/google/oauth2client/issues/252, but it did not get assigned or implemented.

**Comment by jboeuf** [comment]
We know we'll have to update this when we have JWT Access for python but I will merge the current work for now.

**Pull request merged**